### ISSUE TITLE
configure section for libev looks for event.h 

### ISSUE THREAD

**Issue by NateCrawford** [openpmix/openpmix#3258]
### What version of the PMIx Reference Library are you using? 
4.2.8

### Describe how PMIx was installed 
tarball

### Please describe the system on which you are running

* Operating system/version:
Rocky 8.9
* Computer hardware:
x86_64
* Network type:
Infiniband

-----------------------------

## Details of the problem

When running configure, the libev tests fail when they can't find event.h. The header file provided by libev-devel is ev.h, and we don't have the corresponding libevent-devel package installed.

I'm guessing it's a typo, as manually editing the configure script to test for ev.h completes as expected.

Here? --> https://github.com/openpmix/openpmix/blob/00d43d2191464d3ac7f452143190e274f8df91d2/config/pmix_setup_libev.m4#L66

**Comment by NateCrawford** [comment]
In this case compilation is still blocked by a number of files still including event.h.

For now, dropping back to use libevent instead of libev.

**Comment by rhc54** [comment]
What version of libev are you using? Sounds like he made a change as there has been no previous problem with using it.

**Comment by NateCrawford** [comment]
The package installed from the EL8 OS "appstream" repo is libev-devel-4.24-6.el8.x86_64. It provides:
/usr/include/ev++.h
/usr/include/ev.h
/usr/lib64/libev.so
/usr/share/man/man3/ev.3.gz

4.24 was originally released in 2016, and can provide its own version of event.h as a compatibility wrapper. This would, however, conflict with libevent-devel's event.h, so I assume the packagers left that out. I'll look into it.

If you compile libev and install somewhere else, there's probably no issue.

Can pmix be modified to use ev.h directly when libev is used?

**Comment by NateCrawford** [comment]
And a little more searching gets this package in the repo: libev-libevent-devel-4.24-6.el8.x86_64, which contains the missing event.h wrapper. It obviously conflicts with the libevent-devel package, so there would still be a small benefit to having pmix not use event.h if libev is chosen. 

Testing config and build with libev now.

**Comment by NateCrawford** [comment]
Config and build with the wrapper event.h work, but the self-tests hang and fail with a timeout. That, however, would be a different issue, and this one can probably be closed.

For now, going with libevent-2.1.8-5.el8

**Comment by rhc54** [comment]
I don't know the last time someone used libev - no real benefit to doing so over libevent. We only added that support because someone was using a package based on libev, and there are too many conflicts if you try to link against both event libraries. Haven't heard anything recently about it, to be honest.

I'll leave this open so we can look at it, but it will be a low priority so I cannot say when we might get to it.

**Comment by NateCrawford** [comment]
Sounds good to me. I did not know libev existed until recently, and don't need to use it. If libevent is used more widely, I'll go with that.

If you do start to look at it, and can't reproduce the tests hanging when using libev, let me know. 

Thanks!

**Comment by rhc54** [comment]
Spent a little time on this and think I'm going to have to abandon libev support entirely. If I try to deal with `ev.h` everything fails to compile - you must have the `event.h` compatibility file or else we need to rewrite a lot of code. If I add that back, then we fail to build anyway because of a reported error in `ev.h`:
```
/Users/rhc/libev/build/v4.33/include/ev.h:244:3: error: expected identifier before numeric constant
  244 |   EV_ERROR    = (int)0x80000000  /* sent when an error occurs */
```
So it looks like something is simply not quite right, and honestly doesn't feel worth the effort. If someone wants libev support re-enabled, then they will need to provide the necessary PR to make it all work again. I'll record the support removal PR here so they can start by reverting it.

**Comment by rhc54** [comment]
Here is the PR to remove support: https://github.com/openpmix/openpmix/pull/3410
